### DISCUSSION TITLE
How to disable self-profiling?

### DISCUSSION THREAD

**Discussion by ogxd** [grafana/pyroscope#2925] [Q&A] [answered]
Hello,

I have deployed an all-in-one instance of Pyroscope, however it seems it profiles itself (`pyroscope` and `grafana-agent`).   
<img width="813" alt="image" src="https://github.com/grafana/pyroscope/assets/12692438/90ad85a3-be1c-4b45-b84b-2c2bdbf3a3a3">

I would like to get rid of this and only have my application profiling data. It seemed to me like `self_profiling.disable_push: true` was the way to go but it seems it either does not work or isn't what I am looking for

From the documentation:
```yml
self_profiling:
  # When running in single binary (--target=all) Pyroscope will push (Go SDK)
  # profiles to itself. Set to true to disable self-profiling.
  # CLI flag: -self-profiling.disable-push
  [disable_push: <boolean> | default = false]
```

My configuration (from `/api/v1/status/config`)
<img width="263" alt="image" src="https://github.com/grafana/pyroscope/assets/12692438/ce7bfcaa-61f5-4e3c-a58a-9ed6be53400d">

I used the pyroscope helm chart 1.3.4 for deployment. The service version is 1.2.1.

Thanks in advance for your help!

**Comment by korniltsev** [comment]
your configuration is correct

This disables self profiling for my instance
```
self_profiling:
  disable_push: false
```

**Reply by ogxd** [reply]
My configuration is `disable_push: true`. Do you mean I must put  `disable_push: false` to disable self profiling ?

**Reply by korniltsev** [reply]
Im sorry , it should have been 

```
self_profiling:
  disable_push: true
```
Your configuration seem correct

**Reply by ogxd** [reply]
Then it's weird.. do you also use `target: all` ? And do you use the helm chart ? https://artifacthub.io/packages/helm/grafana/pyroscope

**Reply by korniltsev** [reply]
yes. Could you share you agent configuration? I wonder if it is the agent scraping pyroscope pods.

**Reply by ogxd** [reply]
I don't use the Grafana Agent, I use this profiler native library that gets loaded by the dotnet runtime: https://github.com/grafana/pyroscope-dotnet

I use it by copying the profiler library along my app in the docker file, like this:
```
RUN curl -s -L https://github.com/grafana/pyroscope-dotnet/releases/download/v0.8.14-pyroscope/pyroscope.0.8.14-glibc-x86_64.tar.gz  | tar xvz -C .
COPY --from=pyroscope/pyroscope-dotnet:0.8.14-glibc /Pyroscope.Profiler.Native.so ./Pyroscope.Profiler.Native.so
COPY --from=pyroscope/pyroscope-dotnet:0.8.14-glibc /Pyroscope.Linux.ApiWrapper.x64.so ./Pyroscope.Linux.ApiWrapper.x64.so
ENV CORECLR_ENABLE_PROFILING=1
ENV CORECLR_PROFILER={BD1A650D-AC5D-4896-B64F-D6FA25D6B26A}
ENV CORECLR_PROFILER_PATH=/app/Pyroscope.Profiler.Native.so
ENV LD_PRELOAD=/app/Pyroscope.Linux.ApiWrapper.x64.so
ENV PYROSCOPE_APPLICATION_NAME=myservice
ENV PYROSCOPE_PROFILING_ENABLED=1
ENV PYROSCOPE_PROFILING_CPU_ENABLED=1
ENV PYROSCOPE_PROFILING_EXCEPTION_ENABLED=1
ENV PYROSCOPE_PROFILING_ALLOCATION_ENABLED=1
ENV PYROSCOPE_PROFILING_LOCK_ENABLED=1
ENV PYROSCOPE_PROFILING_WALLTIME_ENABLED=0
```

Still, there must be a Grafana Agent somewhere 🤔 
<img width="341" alt="image" src="https://github.com/grafana/pyroscope/assets/12692438/33f60cfe-7466-4533-b929-6fe2aea1216c">

Here is the full configuration from `/api/v1/status/config`:
[config.yml](https://github.com/grafana/pyroscope/files/13950888/config.json) (.json but it's yml because github does not allow yml files for some reason)

**Reply by korniltsev** [reply, marked as answer]
https://github.com/grafana/pyroscope/blob/ed65138dacb5610595e63f7c19211f5f46df6a40/operations/pyroscope/helm/pyroscope/values.yaml#L180

**Reply by ogxd** [reply]
It worked ! Thanks a lot 🫶
